packages/expo-system-ui/android/src/main/java/expo/modules/systemui/SystemUIModule.kt

```diff
@@ -0,0 +1,262 @@
+    activity.runOnUiThread {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        activity.window.navigationBarDividerColor = color
+        promise.resolve(null)
+      } else {
+        promise.reject("unavailable", "'setNavigationBarDividerColor' is only available on Android API 28 or higher!")
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun getStatusBarBackgroundColor(promise: Promise) {
+    activity.runOnUiThread {
+      val color = colorToHex(activity.window.statusBarColor)
+      promise.resolve(color)
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarBackgroundColor(color: Int, promise: Promise) {
+    activity.runOnUiThread {
+      activity.window.statusBarColor = color
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun getStatusBarForegroundStyle(promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        val style = if (controller.isAppearanceLightStatusBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarForegroundStyle(style: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (style) {
+          "light" -> controller.isAppearanceLightStatusBars = false
+          "dark" -> controller.isAppearanceLightStatusBars = true
+          else -> {
+            promise.reject("invalid-value", "Value \"$style\" is not a valid Status Bar foreground style!")
+            return@let
+          }
+        }
+        promise.resolve(null)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun getNavigationBarForegroundStyle(promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        val style = if (controller.isAppearanceLightNavigationBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarForegroundStyle(style: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (style) {
+          "light" -> controller.isAppearanceLightNavigationBars = false
+          "dark" -> controller.isAppearanceLightNavigationBars = true
+          else -> {
+            promise.reject("invalid-value", "Value \"$style\" is not a valid Navigation Bar foreground style!")
+            return@let
+          }
+        }
+        promise.resolve(null)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarVisibility(visibility: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (visibility) {
+          "visible" -> {
```

Comment:
Are there more visibility types, or can this be binary?

Reply:
Seems there are, maybe we want to map this to the app.json values https://docs.expo.dev/versions/latest/config/app/#androidnavigationbar or sketch up a plan to change the app.json values to better match the dynamic API. CC @brentvatne

Reply:
Oh I didn't know that we had those visibility values! I used TS unions here in case a type in the middle of hidden and visible comes along sometime, e.g. `translucent`.